crates/autopilot/src/run_loop.rs

```diff
@@ -379,4 +391,24 @@ impl RunLoop {
             .await
             .context("save competition data")
     }
+
+    /// Stores [`OrderEventLabel::Considered`] for all orders that were not in
+    /// the winning solution but were proposed in other valid solutions.
+    fn track_considered_orders(
+        &self,
+        winner: &solve::Response,
+        others: &[(usize, solve::Response)],
+    ) {
+        let winning_orders: HashSet<&OrderUid> = winner.orders.iter().collect();
+        let considered_orders: HashSet<&OrderUid> = others
+            .iter()
+            .flat_map(|(_, response)| response.orders.iter())
+            .collect();
+        let considered_orders: Vec<OrderUid> = considered_orders
+            .difference(&winning_orders)
+            .map(|o| **o)
+            .collect();
+        self.database
+            .store_order_events(considered_orders, OrderEventLabel::Considered);
+    }
```

Comment:
One way to avoid the issue with `considered XOR executing` that you mentioned would be to label all orders included in solutions as `considered`, and then when we are going to execute them, then add an `executing` event.

This way we don't have such an issue with missing data, and an order for which there was a proposed solution that did not execute (regardless of the reason, whether it was second place or because of an error in between choosing the winner and executing) will get an event.

Reply:
My initial inclination was to that as well but in the issue you were pretty specific:
> 'considered': the order was included in a settlement that did not win

But since you are fine with relaxing that constraint a bit I'm happy to simplify the code. 👍

Reply:
[906404e](https://github.com/cowprotocol/services/pull/1641/commits/906404ee1812468c1d02cadc8fdf1239636cc41d)

Reply:
Yeah, sorry for moving the goal posts, just thought it made sense given how the code turned out.